tools/section-link-check.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+for file in `find . -name "*.md" -type f`; do
```

Comment:
Do we want to do this for all markdown files? I'd rather do it only for the specifications, because those are the one that people might reference with links in source code.

Please extract the body of the loop in a function and, for now,  apply it only to the 4 markdown of the specs. If we ever decide that it is worth running it on all markdown files, we just need to add the loop.